### PR TITLE
HBASE-23556: Minor ChoreService Cleanup

### PR DESCRIPTION
**Pull request by belugabehr**

### CODE REVIEW

File: hbase-common/src/main/java/org/apache/hadoop/hbase/ChoreService.java [hunk 1]
```diff
@@ -173,9 +173,7 @@ public synchronized boolean scheduleChore(ScheduledChore chore) {
    * @param chore The Chore to be rescheduled. If the chore is not scheduled with this ChoreService
    *          yet then this call is equivalent to a call to scheduleChore.
    */
-  private synchronized void rescheduleChore(ScheduledChore chore) {
-    if (chore == null) return;
-
```
Comment (belugabehr) [inline hbase-common/src/main/java/org/apache/hadoop/hbase/ChoreService.java]: This is a private method.  The public methods that call this are already synchronized and already ensure that 'chore' is not null.

File: hbase-common/src/main/java/org/apache/hadoop/hbase/ChoreService.java [hunk 2]
```diff
@@ -352,17 +349,14 @@ public boolean isTerminated() {
...
     // keySet the results would be undefined because the keySet would be changing
-    for (ScheduledChore chore : scheduledChores.keySet()) {
-      choresToCancel.add(chore);
-    }
+    ArrayList<ScheduledChore> choresToCancel = new ArrayList<>(scheduledChores.keySet());
+
     for (ScheduledChore chore : choresToCancel) {
       cancelChore(chore, mayInterruptIfRunning);
     }
-    choresToCancel.clear();
```
Comment (belugabehr) [inline hbase-common/src/main/java/org/apache/hadoop/hbase/ChoreService.java]: No need to `clear` this data structure.  All entries will be cleaned up once the `ArrayList` itself is.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   3m 51s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -0 :warning: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ master Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   6m  2s |  master passed  |
| +1 :green_heart: |  compile  |   0m 22s |  master passed  |
| +1 :green_heart: |  checkstyle  |   0m 28s |  master passed  |
| +1 :green_heart: |  shadedjars  |   5m  5s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   0m 21s |  master passed  |
| +0 :ok: |  spotbugs  |   0m 51s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |   0m 49s |  master passed  |
||| _ Patch Compile Tests _ |
| +1 :green_heart: |  mvninstall  |   5m 30s |  the patch passed  |
| +1 :green_heart: |  compile  |   0m 23s |  the patch passed  |
| +1 :green_heart: |  javac  |   0m 23s |  the patch passed  |
| +1 :green_heart: |  checkstyle  |   0m 26s |  hbase-common: The patch generated 0 new + 3 unchanged - 1 fixed = 3 total (was 4)  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  shadedjars  |   5m  1s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |  17m 19s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2.  |
| +1 :green_heart: |  javadoc  |   0m 20s |  the patch passed  |
| +1 :green_heart: |  findbugs  |   0m 55s |  the patch passed  |
||| _ Other Tests _ |
| +1 :green_heart: |  unit  |   2m 57s |  hbase-common in the patch passed.  |
| +1 :green_heart: |  asflicense  |   0m 12s |  The patch does not generate ASF License warnings.  |
|  |   |  56m 51s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.5 Server=19.03.5 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-927/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/927 |
| JIRA Issue | HBASE-23556 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile |
| uname | Linux 9a7c84929202 4.15.0-66-generic #75-Ubuntu SMP Tue Oct 1 05:24:09 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-927/out/precommit/personality/provided.sh |
| git revision | master / a580b1d2e9 |
| Default Java | 1.8.0_181 |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-927/1/testReport/ |
| Max. process+thread count | 259 (vs. ulimit of 10000) |
| modules | C: hbase-common U: hbase-common |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-927/1/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by busbey** [comment]
In the future can we leave some more time on these kind of refactoring reviews? especially when it involves changes that can impact correctness (removing synchronization) or gc (removing proactive nulling out of references)?

**Pull request merged**